File: test-api.js
import http from 'http';

const postData = JSON.stringify({});

const options = {
  hostname: 'localhost',
  port: 3000,
  path: '/analyze',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(postData)
  }
};

const req = http.request(options, (res) => {
  console.log(`STATUS: ${res.statusCode}`);
  console.log(`HEADERS:`, JSON.stringify(res.headers));
  
  let data = '';
  
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    console.log('RESPONSE BODY:');
    console.log(data);
    process.exit(0);
  });
});

req.on('error', (e) => {
  console.error(`Problem with request: ${e.message}`);
  process.exit(1);
});

// Write data to request body
req.write(postData);
req.end();

// Add timeout
setTimeout(() => {
  console.error('Request timed out after 10 seconds');
  process.exit(1);
}, 10000);

File: test_api.py
import requests
import json

url = "http://localhost:3000/analyze"

payload = {
    "prFiles": [
        {
            "filename": "src/app.js",
            "patch": "@@ -1,3 +1,4 @@\nimport axios from 'axios';\n+const secret = 'sk-12345';\nconsole.log('hello');"
        }
    ],
    "architectRules": {
        "project": "test-app",
        "layers": {
            "frontend": {
                "allowedImports": ["react"],
                "forbiddenImports": ["secrets"]
            }
        }
    }
}

print("Testing /analyze endpoint...")
try:
    response = requests.post(url, json=payload, timeout=20)
    print(f"Status: {response.status_code}")
    try:
        print(json.dumps(response.json(), indent=2))
    except:
        print(response.text[:500])
except requests.Timeout:
    print("Timeout after 20 seconds")
except Exception as e:
    print(f"Error: {e}")
